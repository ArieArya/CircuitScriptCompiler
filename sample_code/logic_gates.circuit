print(and(1, 1) == 1);
print(and(0, 1) == 0);

print(or(0, 1) == 1);
print(or(0, 0) == 0);

print(not(0) == 1);

print(xor(0, 1) == 1);
