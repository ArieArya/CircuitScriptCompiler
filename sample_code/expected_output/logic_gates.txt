Lexing file: logic_gates.circuit
Tokens:
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.KEYWORD (Lexeme = 'and')
Token.LPAREN (Lexeme = '(')
Token.DIGIT (Lexeme = '1')
Token.COMMA (Lexeme = ',')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.OPERATOR (Lexeme = '==')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.KEYWORD (Lexeme = 'and')
Token.LPAREN (Lexeme = '(')
Token.DIGIT (Lexeme = '0')
Token.COMMA (Lexeme = ',')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.OPERATOR (Lexeme = '==')
Token.DIGIT (Lexeme = '0')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.KEYWORD (Lexeme = 'or')
Token.LPAREN (Lexeme = '(')
Token.DIGIT (Lexeme = '0')
Token.COMMA (Lexeme = ',')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.OPERATOR (Lexeme = '==')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.KEYWORD (Lexeme = 'or')
Token.LPAREN (Lexeme = '(')
Token.DIGIT (Lexeme = '0')
Token.COMMA (Lexeme = ',')
Token.DIGIT (Lexeme = '0')
Token.RPAREN (Lexeme = ')')
Token.OPERATOR (Lexeme = '==')
Token.DIGIT (Lexeme = '0')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.KEYWORD (Lexeme = 'not')
Token.LPAREN (Lexeme = '(')
Token.DIGIT (Lexeme = '0')
Token.RPAREN (Lexeme = ')')
Token.OPERATOR (Lexeme = '==')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
Token.KEYWORD (Lexeme = 'print')
Token.LPAREN (Lexeme = '(')
Token.KEYWORD (Lexeme = 'xor')
Token.LPAREN (Lexeme = '(')
Token.DIGIT (Lexeme = '0')
Token.COMMA (Lexeme = ',')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.OPERATOR (Lexeme = '==')
Token.DIGIT (Lexeme = '1')
Token.RPAREN (Lexeme = ')')
Token.SEMICOLON (Lexeme = ';')
